android/titanium/src/java/org/appcelerator/titanium/io/TiResourceFile.java

```diff
@@ -193,6 +192,19 @@ public long size()
 		InputStream is = null;
 		try {
 			is = getInputStream();
+
+			// CipherInputStream.available() always returns 0
+			// Iterate through stream to obtain true size.
+			if (is instanceof CipherInputStream) {
+				long size = 0;
+				long read = 0;
```

Comment:
technically this should be an int, as #read() returns an int.

Reply:
Ah, forgot to amend this after switching from `.skip()`